.github/workflows/build-to-cloudflare-pages.yml

```diff
@@ -0,0 +1,124 @@
+                  ref: ${{ github.event.pull_request.head.sha }}
+
+            - name: Setup node
+              uses: actions/setup-node@v2
+
+            - name: Generate build status comment
+              id: build_status_comment
+              uses: actions/github-script@v3
+              with:
+                  script: |
+                      const preview_url = "https://" + "${{ github.head_ref }}" + ".deriv-com.pages.dev"
+                      const comment = [
+                        '## Cloudflare Pages Preview link',
+                        '| Name | Result |',
+                        '| :--- | :------ |',
+                        `| **Build status**  | In progress ⏳ |`,
+                        `| **Preview URL**  | [Visit Preview](${preview_url}) |`,
+                        ''
+                      ].join('\n')
+                      core.setOutput("comment", comment);
+
+            - name: Update build status comment
+              uses: marocchino/sticky-pull-request-comment@v2
+              with:
+                  only_update: true
+                  header: Cloudflare Pages Preview Comment
+                  number: ${{github.event.issue.number}}
+                  message: ${{steps.build_status_comment.outputs.comment}}
+
+            - name: Get build output from master cache
+              uses: actions/cache/restore@v3
+              with:
+                  key: master-cache-public
+                  restore-keys: |
+                      master-cache-public-replica
+                  path: |
+                      .cache
+                      public
+
+            - name: Get cached dependencies
+              id: cache-npm
+              uses: actions/cache/restore@v3
+              with:
+                  path: node_modules
+                  key: npm-${{ hashFiles('**/package-lock.json') }}
+
+            - name: Install dependencies
+              if: ${{ steps.cache-npm.outputs.cache-hit != 'true' }}
+              run: npm ci
+
+            - name: Build project
+              id: build-project
+              run: npm run build
+
+            - name: Publish to Cloudflare Pages
+              id: publish-to-pages
+              run: |
+                  echo "Installing Wrangler CLI"
+                  npm i -g wrangler@3.5.0
+
+                  echo "Deploying build to Cloudflare Pages"
+                  directory='public'
+                  projectName='deriv-com'
+                  branch=$GITHUB_HEAD_REF
+
+                  preview_url=$(wrangler pages deploy $directory --project-name=$projectName --branch=$branch > log.txt 2>&1; echo $?)
+                  echo "------"
+                  cat log.txt
+
+                  branchName=$(echo $branch | sed 's/[\/_]/-/g')
+                  if grep -q "Deployment complete" log.txt; then
+                    echo "preview_url=https://$branchName.deriv-com-playground.pages.dev" > "$GITHUB_OUTPUT"
+                  else
+                    exit 1
```

Comment:
Can we add a message, "deployment failed" ?

Reply:
Sure I can an additional message, but the `log.txt` is just used to log the progress of running `wrangler pages deploy`, because the Github Actions does not log out the stdout of `wrangler` 😓 , for instance:
### Successful build
*Before adding log file*
![Screenshot 2023-08-16 at 1 42 11 PM](https://github.com/binary-com/deriv-com/assets/103016120/44668dd6-4abf-440c-8163-2a0520b8f277)

*After adding log file*
![Screenshot 2023-08-16 at 1 40 25 PM](https://github.com/binary-com/deriv-com/assets/103016120/f3dd1284-b6f8-433a-bb2a-8eed7ab76e39)

This is especially useful during debugging deployment errors:
*Before adding log file*
![Screenshot 2023-08-16 at 1 43 45 PM](https://github.com/binary-com/deriv-com/assets/103016120/67efe583-f157-4d1c-a7d6-aafd22a82e20)

*After adding log file*
![Screenshot 2023-08-16 at 1 46 13 PM](https://github.com/binary-com/deriv-com/assets/103016120/76f8c9b4-d80d-4426-8c52-4e8242c1d6d8)

Reply:
Cool, let's just add the message before exit. LGTM